tools/pubsys/src/aws/promote_ssm/mod.rs

```diff
@@ -216,8 +231,86 @@ pub(crate) async fn run(args: &Args, promote_args: &PromoteArgs) -> Result<()> {
     Ok(())
 }
 
+/// Read parameters in given file, add newly promoted parameters, and write combined parameters to
+/// the given file
+async fn write_rendered_parameters(
+    ssm_parameters_output: &PathBuf,
+    set_parameters: &HashMap<SsmKey, String>,
+    source_target_map: HashMap<&String, &String>,
+) -> Result<()> {
+    info!(
+        "Writing promoted SSM parameters to {}",
+        ssm_parameters_output.display()
+    );
+    let parsed_parameters = parse_expected_parameters(&ssm_parameters_output.to_owned())
+        .await
+        .context(error::ParseExistingSsmParametersSnafu {
+            path: ssm_parameters_output,
+        })?;
+
+    let combined_parameters: HashMap<Region, HashMap<SsmKey, String>> =
+        combine_parameters(parsed_parameters, set_parameters, source_target_map);
+
+    serde_json::to_writer_pretty(
```

Comment:
Should this be a function somewhere? We write out this parameter data structure in more than one place now.